src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -39,7 +39,7 @@ internal sealed class EditAndContinueWorkspaceService : IEditAndContinueWorkspac
         private readonly EditAndContinueDiagnosticUpdateSource _emitDiagnosticsUpdateSource;
         private readonly EditSessionTelemetry _editSessionTelemetry;
         private readonly DebuggingSessionTelemetry _debuggingSessionTelemetry;
-        private readonly ICompilationOutputsProviderService _compilationOutputsProvider;
+        private readonly Func<Project, CompilationOutputs> _compilationOutputsProvider;
```

Comment:
not a huge fan of passing around a delegate like this.  Doesn't seem to match our general patterns here.  Even if it's a single method interface, that seems pretty clear to me.

Reply:
This is simpler than defining an interface and implementing a class, and it's equivalent. We already have `Action<DebuggingSessionTelemetry.Data> _reportTelemetry` right below it.

Reply:
Agreed with @CyrusNajmabadi an interface seems appropriate.

Reply:
I don't think we should need an interface just to implement a test hook.